TempleEight/docs/guide/orchestration.md

```diff
@@ -3,3 +3,169 @@ id: orchestration
+This file performs all of the steps needed to get everything running correctly, including runnning the `configure-kong.sh` script and setting the `$KONG_ENTRY` and `$KONG_ADMIN` environment variables.
+In order for these variables to remain set for the remainder of your terminal session, it needs to be run with the `source` command.
+
+Assuming you have the Docker daemon up and running, lets spin up our application:
+
+```shell 
+❯❯❯ source deploy.sh
+```
+
+A lot of output will be presented, detailing each step of the deployment process, and any errors that occurred. 
+
+Verify the system was configured correctly by checking the environment variables:
+
+```shell
+❯❯❯ echo $KONG_ENTRY 
+localhost:8000
+
+❯❯❯ echo $KONG_ADMIN
+localhost:8001
+```
+
+At this point, you're free to make requests to your services.
+In order to access them, address requests to Kong's ingress URL: `$KONG_ENTRY/api/{service-name}/{entity-id}` and Kong will forward the request to the right place.
+
+Once you are finished with your infrastructure, everything can be cleanly shut down with:
+
+```shell
+❯❯❯ docker-compose down
+```
+
+## Kubernetes
+
+Another common framework for orchestrating services is [Kubernetes](https://kubernetes.io/)(kube, k8s). 
+It's a more heavyweight tool than Docker Compose, but has much more mature features and is generally seen as more 'production grade'.
+
+By changing the `#provider` annotation in a Templefile's project block to read `#provider(kubernetes)`, 
+Temple will generate the required config to run your services in k8s. 
+
+Changing the `example.temple` file from the [Getting Started](../getting-started) guide to use this `#provider` annotation and generating,
+the configuration files generated are:
+
+```shell
+.
+├── deploy.sh
+├── kong
+│   └── configure-kong.sh
+├── kube
+│   ├── deploy
+│   │   ├── deploy-daemon-set.yaml
+│   │   ├── deploy-replication-controller.yaml
+│   │   └── deploy-service.yaml
+│   ├── example-service
+│   │   ├── db-deployment.yaml
+│   │   ├── db-service.yaml
+│   │   ├── db-storage.yaml
+│   │   ├── deployment.yaml
+│   │   └── service.yaml
+│   └── kong
+│       ├── kong-db-deployment.yaml
+│       ├── kong-db-service.yaml
+│       ├── kong-deployment.yaml
+│       ├── kong-migration-job.yaml
+│       └── kong-service.yaml
+└── push-image.sh
+```
+
+As you can see, there are a lot more files generated here than for Docker Compose.
+
+The `deploy.sh` and `configure-kong.sh` scripts serve the same purpose here as they do in the Docker Compose example, 
+although the mechanisms used to achieve this are different.
+
+The `push-image.sh` script builds Docker images from each service generated in your project.
+Since Kubernetes requires all images used in your system to be hosted in a [Docker Registry](https://docs.docker.com/registry/), Temple's Kubernetes infrastructure features it's own registry, hosted in the cluster itself.
```

Comment:
```suggestion
Since Kubernetes requires all images used in your system to be hosted in a [Docker Registry](https://docs.docker.com/registry/), Temple's Kubernetes infrastructure features its own registry, hosted in the cluster itself.
```